projects/packages/jetpack-mu-wpcom/src/class-jetpack-mu-wpcom.php

```diff
@@ -521,4 +522,118 @@ public static function load_social_links() {
 			new \Automattic\Jetpack\Classic_Theme_Helper\Social_Links();
 		}
 	}
+
+	/**
+	 * Checks for translation updates and installs them if necessary.
+	 */
+	public static function maybe_update_translations() {
+		$plugin_slug = 'jetpack-mu-wpcom';
+
+		// Get installed site locales.
+		$locales = self::get_all_active_locales();
+		if ( empty( $locales ) ) {
+			return;
+		}
+
+		// Query the translation API.
+		$api_url      = 'https://translate.wordpress.com/api/translations-updates/wpcom/plugins';
+		$request_body = array(
+			'locales' => $locales,
+			'plugins' => array(
+				$plugin_slug => array( 'version' => 'latest' ),
+			),
+		);
+
+		$response = wp_remote_post(
```

Comment:
Will this wind up performing this fetch on every page view on the site? Is that going to be a performance concern, for the site itself and for translate.wordpress.com?

Reply:
The goal of this was to run the fetch once the plugin was loaded/activated, not on every page request! I can add caching to prevent this, but I think using transients with a scheduled cron job might be a better alternative.